commons-macros/src/main/scala/com/avsystem/commons/macros/MacroCommons.scala

```diff
@@ -1409,17 +1409,26 @@ trait MacroCommons extends CompatMacroCommons { bundle =>
       }
     }
 
-  def determineTypeParams(undetTpe: Type, detTpe: Type, typeParams: List[Symbol]): Option[List[Type]] = {
+  def determineTypeParams(undetTpe: Type, detTpe: Type, typeParams: List[Symbol], debug: Boolean = false): Option[List[Type]] = {
     val methodName = c.freshName(TermName("m"))
     val typeDefs = typeParams.map(typeSymbolToTypeDef(_, forMethod = true))
 
+    if(debug) {
+        q"""
+        def $methodName[..$typeDefs](f: ${treeForType(undetTpe)} => $UnitCls): $UnitCls = ()
+        $methodName((_: $detTpe) => ())
+      """.debug("")
+    }
+
```

Comment:
committed intentionally?

Reply:
Of course not, I was too happy it started working :D